### PR TITLE
Feature/optional secondary

### PR DESCRIPTION
**Pull request by bogdang989**
Pull request for the issue https://github.com/common-workflow-language/common-workflow-language/issues/717

### CODE REVIEW

File: Process.yml [hunk 1]
```diff
@@ -460,16 +460,15 @@ $graph:
         - "null"
         - string
         - Expression
+        - SecondaryFileSchema
         - type: array
-          items: [string, Expression]
+          items: [string, Expression, SecondaryFileSchema]
```
Comment (tetron) [inline Process.yml]: Add "null" to items list.

File: Process.yml [hunk 2]
```diff
@@ -850,3 +849,44 @@ $graph:
...
+- name: SecondaryFileSchema
+  type: record
+  fields:
+    - name: pattern
+      type:
+        - string
+        - Expression
+      doc: |
+        Provides a pattern or expression specifying files or directories that
+        must be included alongside the primary file.
```
Comment (tetron) [inline Process.yml]: "should be"

File: Process.yml [hunk 3]
```diff
@@ -850,3 +849,44 @@ $graph:
...
+
+        If a value in `secondaryFiles` is a string that is not an expression,
+        it specifies that the following pattern should be applied to the path
+        of the primary file to yield a filename relative to the primary File:
+
+          1. If string begins with one or more caret `^` characters, for each
+            caret, remove the last file extension from the path (the last
+            period `.` and all following characters).  If there are no file
+            extensions, the path is unchanged.
+          2. Append the remainder of the string to the end of the file path.
```
Comment (tetron) [inline Process.yml]: Document "?" syntax for optionals

File: Process.yml [hunk 4]
```diff
@@ -850,3 +849,44 @@ $graph:
...
+
+        If the value is an expression, the value of `self` in the expression
+        must be the primary input or output File object to which this binding
+        applies.  The `basename`, `nameroot` and `nameext` fields must be
+        present in `self`.  For `CommandLineTool` outputs the `path` field must
+        also be present.  The expression must return a filename string relative
+        to the path to the primary File, a File or Directory object with either
+        `path` or `location` and `basename` fields set, or an array consisting
+        of strings or File or Directory objects.  It is legal to reference an
+        unchanged File or Directory object taken from input as a secondaryFile.
```
Comment (tetron) [inline Process.yml:886]: The expression may return "null" in which case there is no secondaryFile from that expression.

File: tests/docker-array-secondaryfiles.cwl [hunk 5]
```diff
@@ -16,7 +16,10 @@ inputs:
       type: array
       items: File
     secondaryFiles:
-      - .fai
+      - pattern: .fai
+        required: true
+      - pattern: .blablai
+        required: false
```
Comment (tetron) [inline tests/docker-array-secondaryfiles.cwl:22]: Add test case for "?" syntax, and another test case with an expression which conditionally returns "null"

**Review by tetron — CHANGES_REQUESTED** [review]

File: Process.yml (hunk 4 above)
Comment (bogdang989) [reply Process.yml:886]: I added this line, but somehow the conversation was not resolved automatically

**Review by tetron — APPROVED** [review]
